# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/LIR.h
# Commit: 920da79f43fc
# Full Hash: 920da79f43fcf5802ebd5eb2bf610a18e821dacc
# Author: Andy Wingo <wingo@igalia.com>
# Date: 2020-03-19 03:53:34
# Regressor Bug: 1620197
# File Overlap Count: 1
# Description:
#   Bug 1620197 - Enable multiple results from WebAssembly functions r=lth
#   
#   This patch enables multi-value calls and returns, adding some tests, and
#   conditionally disabling a couple expect-fail reftests that now pass.
#   
# ==============================================================================

diff -r 98d7f9e4839f -r 920da79f43fc js/src/jit/LIR.h
--- a/js/src/jit/LIR.h	Wed Mar 18 17:12:14 2020 +0000
+++ b/js/src/jit/LIR.h	Wed Mar 18 16:36:45 2020 +0000
@@ -35,6 +35,7 @@
 class LArgument;
 class LConstantIndex;
 class LInstruction;
+class LDefinition;
 class MBasicBlock;
 class MIRGenerator;
 
@@ -408,7 +409,7 @@
 
   ResultIterator results() const { return ResultIterator(*this); }
 
-  inline LStackSlot resultAlloc(LInstruction* lir) const;
+  inline LStackSlot resultAlloc(LInstruction* lir, LDefinition* def) const;
 };
 
 // Arguments are reverse indices into the stack. The indices are byte indices.